tests/model/test_model.py

```diff
@@ -2624,6 +2624,51 @@ def test__handle_reaction_event_for_msg_in_index(
     def update_message_flags_operation(self, request):
         return request.param
 
+    @pytest.mark.parametrize(
+        "event,initial_alerted_words, expected_alert_words",
+        [
+            (
+                {"type": "alert_words", "alert_words": ["word1", "word2", "word3"]},
+                ["word1"],
+                ["word1", "word2", "word3"],
+            ),
+            (
+                {"type": "alert_words", "alert_words": []},
+                ["word1"],
+                [],
+            ),
+            (
+                {"type": "alert_words", "alert_words": ["word1", "word4"]},
+                ["word1"],
+                ["word1", "word4"],
+            ),
+            (
+                {"type": "alert_words", "alert_words": ["word1", "word2"]},
+                ["word1", "word2", "word3"],
+                ["word1", "word2"],
+            ),
+            (
+                {"type": "alert_words", "alert_words": ["word1", "word3"]},
+                ["word1", "word2"],
+                ["word1", "word3"],
+            ),
+        ],
+        ids=[
+            "Add multiple alert words",
+            "Empty alert words",
+            "Add single new alert words",
+            "Delete an alert word",
+            "Add and delete alert words",
+        ],
+    )
+    def test__handle_alert_words_event(
```

Comment:
I would expect to see this test in the same commit as the new `_handle_*_event` function.

You might like to split the function+test, then connect it up with the event-handler system.

Reply:
I have squashed the 2nd and 3rd commit into one.